### PR TITLE
Several recent updates

### PR DESCRIPTION
**Pull request by fillup**
- add support for network load balancers
- update alb security policy
- add parameter_group_name option for mariadb
- add ecs-service-only-with-volume
- added service-no-alb-with-volume
- add key_name to launch configuration

### CODE REVIEW

**Review by dalenewby — DISMISSED** [review]

File: aws/ecs/service-no-alb-with-volume/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ - `task_role_arn` - ARN for role to assign to task definition. Default: `blank`
+ - `network_mode` - Networking mode for task. Default: `bridge`
+ - `deployment_maximum_percent` - Upper limit of tasks that can run during a deployment. Default: `200`%
+ - `deployment_minimum_healthy_percent` - Lower limit of tasks that must be running during a deployment. Default: `50`%
+
+## Outputs
+
+ - `task_def_arn` - ARN for task definition.
+ - `task_def_family` - Family name of task definition.
+ - `task_def_revision` - Revision number of task definition.
```
Comment (Baggerone) [inline aws/ecs/service-no-alb-with-volume/README.md]: You're missing "task_def_revision_via_data"

File: aws/ecs/service-no-alb-with-volume/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  value = "${aws_ecs_service.service.name}"
+}
+
+output "service_cluster" {
+  value = "${aws_ecs_service.service.cluster}"
+}
+
+output "service_desired_count" {
+  value = "${aws_ecs_service.service.desired_count}"
+}
```
Comment (Baggerone) [inline aws/ecs/service-no-alb-with-volume/outputs.tf]: Did you want to include service_role (as per the readme)?

File: aws/ecs/service-no-alb-with-volume/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+ - `cluster_id` - ID for ECS Cluster
+ - `service_name` - Name of service, all lowercase, no spaces.
+ - `service_env` - Name of environment, used in naming task definition. Ex: `staging`
+ - `container_def_json` - JSON for container definition.
+ - `desired_count` - Number of tasks to run in service
+ - `volume_name` - Name for volume
+ - `volume_host_path` - Path on host EC2 instance to mount volume
+
+### Optional Inputs
```
Comment (Baggerone) [inline aws/ecs/service-no-alb-with-volume/README.md]: Your vars.tf file includes a lot of values that aren't included here.

File: aws/ecs/service-only-with-volume/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ - `container_def_json` - JSON for container definition.
+ - `desired_count` - Number of tasks to run in service
+ - `tg_arn` - Target Group ARN for ALB to register with
+ - `lb_container_name` - Container name from `container_def_json` that should be used with target group / alb
+ - `lb_container_port` - Container port that should be used with target group / alb
+ - `ecsServiceRole_arn` - ARN to IAM ecsServiceRole
+ - `volume_name` - Name for volume
+ - `volume_host_path` - Path on host EC2 instance to mount volume
+
+### Optional Inputs
```
Comment (Baggerone) [inline aws/ecs/service-only-with-volume/README.md]: Your vars.tf includes more inputs than are included here.

File: aws/ecs/service-only-with-volume/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ - `task_role_arn` - ARN for role to assign to task definition. Default: `blank`
+ - `network_mode` - Networking mode for task. Default: `bridge`
+ - `deployment_maximum_percent` - Upper limit of tasks that can run during a deployment. Default: `200`%
+ - `deployment_minimum_healthy_percent` - Lower limit of tasks that must be running during a deployment. Default: `50`%
+
+## Outputs
+
+ - `task_def_arn` - ARN for task definition.
+ - `task_def_family` - Family name of task definition.
+ - `task_def_revision` - Revision number of task definition.
```
Comment (Baggerone) [inline aws/ecs/service-only-with-volume/README.md]: missing `task_def_revision_via_data`

**Comment by dalenewby** [comment]
I copied service-no-alb-with-volume from one of the other modules. Steve's comments would apply to that module as well.

File: aws/ecs/service-no-alb-with-volume/README.md (hunk 1 above)
Comment (fillup) [reply aws/ecs/service-no-alb-with-volume/README.md]: added

File: aws/ecs/service-no-alb-with-volume/outputs.tf (hunk 2 above)
Comment (fillup) [reply aws/ecs/service-no-alb-with-volume/outputs.tf]: removed

File: aws/ecs/service-no-alb-with-volume/README.md (hunk 3 above)
Comment (fillup) [reply aws/ecs/service-no-alb-with-volume/README.md]: removed unused vars

File: aws/ecs/service-only-with-volume/README.md (hunk 4 above)
Comment (fillup) [reply aws/ecs/service-only-with-volume/README.md]: removed unused vars

File: aws/ecs/service-only-with-volume/README.md (hunk 5 above)
Comment (fillup) [reply aws/ecs/service-only-with-volume/README.md]: added

**Comment by fillup** [comment]
I just pushed updates based on @Baggerone's feedback if you want to review again @dalenewby

**Comment by dalenewby** [comment]
Shouldn't the vars.tf files in service-no-alb and service-no-alb-with-volume be more alike?

**Review by Baggerone — DISMISSED** [review]

**Review by dalenewby — APPROVED** [review]

**Pull request merged**